core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/tuple/Tuple.java

```diff
@@ -580,4 +584,56 @@ public Builder removeIfExists(String... attributes) {
         }
 
     }
+
+    public static class BuilderV2 {
+        private final Schema schema;
+        private final Map<String, Object> fieldNameMap;
+
+        public BuilderV2(Schema schema) {
+            this.schema = schema;
+            this.fieldNameMap = new HashMap<>();
+        }
+
+        public BuilderV2 fill(Tuple tuple) {
+            checkNotNull(tuple);
+
+            for (int i = 0; i < tuple.size(); i++) {
+                // TODO There is scope here to be "loose" in the schema matching, so that we don't need a "remove" ever
+                fill(tuple.getSchema().getAttributes().get(i), tuple.getFields().get(i));
+            }
+
+            return this;
+        }
+
+        public BuilderV2 fill(Attribute attribute, Object field) {
+            checkNotNull(attribute);
+            checkAttributeMatchesField(attribute, field);
+
+            if (!schema.containsAttribute(attribute.getName())) {
+                // TODO Create custom exception for Tuple.Builder stuff?
+                throw new RuntimeException(String.format("%s doesn't exist in the expected schema.", attribute.getName()));
```

Comment:
I agree. We can have a separate exception for tuple and schema class. Let's add that in this PR?

Reply:
Yes. I will add this.